### DISCUSSION TITLE
Tasmota scripting diference between chg and upd in the script

### DISCUSSION THREAD

**Discussion by RaphaelSquair** [arendst/Tasmota#20944] [Q&A] [answered]
Hi guys i was questioning myself while programing, what are the diferences in using the chg[var] and the upd[var] ? since both appears to do the same.

**Comment by sfromis** [comment, marked as answer]
Not using scripting, but the wording from the docs suggests that `upd` may be true if updated without being changed, while `chg` is only if actually changed.
> `chg[var]` = true if a variables value was changed
> `upd[var]` = true if a variable was updated

**Reply by RaphaelSquair** [reply]
uhm, this seams to make sense, now i can see some scenarios when this may be useful, thanks for the explanation.
